File: redes-frontend/src/models/DevelopmentFlow.ts
export interface DevelopmentStep {
  order: number;
  component: string;
  description: string;
  communication: string;
  port?: number;
}

export interface OsiParticipation {
  layerNumber: number;
  layerName: string;
  participation: string;
  clarification: string;
}

export interface LogicalPort {
  number: number;
  service: string;
  scope: string;
  purpose: string;
}

export interface DevelopmentFlow {
  applicationName: string;
  description: string;
  steps: DevelopmentStep[];
  osiLayers: OsiParticipation[];
  ports: LogicalPort[];
  technicalNotes: string[];
}

File: redes-backend/src/main/java/com/universidad/redes/domain/model/DevelopmentFlow.java
package com.universidad.redes.domain.model;

import java.util.List;

public record DevelopmentFlow(
        String applicationName,
        String description,
        List<Step> steps,
        List<OsiParticipation> osiLayers,
        List<LogicalPort> ports,
        List<String> technicalNotes
) {

    public DevelopmentFlow {
        if (applicationName == null || applicationName.isBlank()) {
            throw new IllegalArgumentException("El nombre de la aplicación es obligatorio");
        }
        if (description == null || description.isBlank()) {
            throw new IllegalArgumentException("La descripción del flujo es obligatoria");
        }
        steps = List.copyOf(steps);
        osiLayers = List.copyOf(osiLayers);
        ports = List.copyOf(ports);
        technicalNotes = List.copyOf(technicalNotes);
    }

    public record Step(
            int order,
            String component,
            String description,
            String communication,
            Integer port
    ) {
    }

    public record OsiParticipation(
            int layerNumber,
            String layerName,
            String participation,
            String clarification
    ) {
    }

    public record LogicalPort(
            int number,
            String service,
            String scope,
            String purpose
    ) {
    }
}
